### PR TITLE
Docs: Add Datadog guide.

### LINKED ISSUE #13159: Guide to managing Teleport audit logs via Datadog
## Details
See #2922 

### Category
- Improve Existing

### PR DESCRIPTION
**Pull request by alexfornuto**
Resolves #13159
In addition to a new guide on Providing audit events to Datadog, this PR consolidates divergent instructions on installing and configuring the Teleport Event Handler across multiple guides into a single source.

I've also applied some edits to tighten up the nav menu titles for existing export docs.

This branch has structured commits based on the type of change, so they can be reviewed independently.

### CODE REVIEW

**Comment by alexfornuto** [comment]
P.S. The instructions for creating the `teleport-event-handler` role and user, and impersonating that user to create a key could also be consolidated. But I'll leave that for another day.

File: docs/config.json
Comment (zmb3) [inline docs/config.json:575]: Is monitor a better verb than export here?

I ask because monitoring (especially in the context of our docs) is generally referring to Prometheus-style monitoring for performance metrics as opposed to audit events.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (zmb3) [inline docs/pages/management/export-audit-events/datadog.mdx:7]:
```suggestion
how to forward Teleport audit events to Datadog using Fluentd.
```

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (zmb3) [inline docs/pages/management/export-audit-events/datadog.mdx:174]: Looks like this entire block is duplicated in both tabs - would it make sense to move it out of the tabs so we don't risk forgetting to update both sections in tandem in the future?

**Review by zmb3 — APPROVED** [review]

File: docs/config.json
Comment (alexfornuto) [reply docs/config.json:575]: I'm happy with either, I mostly wanted to removing the "ing" suffix. I made the change in words to match the other guides in this list.
Should they all be "Export" over "Monitor"?

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:174]: Yes, but can you think of a good title for the consolidated tab label, to identify the tab as referring to Self-Hosted or Cloud, but not Helm", to contrast the "Helm Chart" tab?

Comment (zmb3) [reply docs/pages/management/export-audit-events/datadog.mdx:174]: Hmm. Nothing comes to mind. This is a tricky one. I've already approved though, so if we don't think of a good option feel free to merge as-is.

Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:174]: $5 says @ptgott has got a great idea when he comes around to reviewing this one.

File: docs/pages/includes/configure-event-handler.mdx
Comment (mdwn) [inline docs/pages/includes/configure-event-handler.mdx:26]:
```suggestion
`TELEPORT_CLUSTER_ADDRESS` to the DNS name and port of your Teleport Auth
```

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (mdwn) [inline docs/pages/management/export-audit-events/datadog.mdx:225]: I'm not 100% sure using `id -u root` and `id -g root` is guaranteed to work on all systems. It likely will, but it assumes the ids associated with root on the host system match those on the Docker container.

That being said, I'd think they likely will, so I'll leave it up to you whether this is worth pursuing.

**Review by mdwn — APPROVED** [review]
Two small comments, otherwise LGTM.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:225]: I'd like to put a comment line at the top of the code block about this, but I'm unsure how to phrase it.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (ptgott) [inline docs/pages/management/export-audit-events/datadog.mdx:391]: It might be good to add a quick section about visualizing audit event counts by the "event" field, e.g., in the Splunk guide (https://goteleport.com/docs/management/export-audit-events/splunk/?scope=enterprise#step-44-visualize-your-audit-logs-in-splunk)

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (ptgott) [inline docs/pages/management/export-audit-events/datadog.mdx:241]: Might be worth noting that the `ca_path`, `cert_path` and `private_key_path` need to match the paths to the credentials you generated in `fluent.conf`.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (ptgott) [inline docs/pages/management/export-audit-events/datadog.mdx:393]: I got "connection reset by peer" errors when running the Event Handler:

```
ERRO   Error sending event to Teleport: Post "https://localhost:8888/test.log": read tcp 127.0.0.1:39460->127.0.0.1:8888: read: connection reset by peer event-handler/app.go:118
```

Looks like this has to do with the TLS handshake between the Event Handler and Fluentd, as reproduced via `curl`:

```
$ curl --key client.key --cert client.crt --cacert ca.crt -v https://localhost:8888/test.log
*   Trying 127.0.0.1:8888...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8888 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: ca.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* OpenSSL SSL_connect: Connection reset by peer in connection to localhost:8888
* Closing connection 0
curl: (35) OpenSSL SSL_connect: Connection reset by peer in connection to localhost:8888
```

Weirdly, the client cert's `Not After` is before its `Not Before`:

```
~$ openssl x509 -text -in client.crt
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            8a:82:e4:d5:f1:7b:c4:e8:a9:56:8e:ee:64:7f:87:7c
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = US, CN = localhost
        Validity
            Not Before: Feb 23 23:22:46 2023 GMT
            Not After : Feb 20 23:22:46 2033 GMT
```

I built the Event Handler from commit `bc85392dc3b03bedcd963709d3ef4bba2af37b2d`.

File: docs/pages/management/export-audit-events/elastic-stack.mdx
Comment (ptgott) [inline docs/pages/management/export-audit-events/elastic-stack.mdx:125]: This table includes some Logstash-specific context that is lost in the partial

File: docs/pages/management/export-audit-events/splunk.mdx
Comment (ptgott) [inline docs/pages/management/export-audit-events/splunk.mdx:132]: This table includes some Splunk-specific context that is lost in the new partial

**Review by ptgott — COMMENTED** [review]
This should also change `docs/pages/management/export-audit-events.mdx` to include a listing for this guide.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:65]:
```suggestion
## Step 1/6. Install the Event Handler plugin
```

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:70]: Resolving in the partial.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:59]: >  Is there a recommended installation method for following this guide?
I went back and forth on how to phrase this in the original writing. I think we want to avoid telling readers the best way to set up 3rd party systems in unknown environments. I'm adding a line that the guide assumes a local docker container for testing.

> Also, do we expect Fluentd to be listening on a specific port for this guide to work?

Yes, and the config file we generate and integrate in the steps below ensures that this port is listening.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:99]: We should probably update all the other guides as well, since this was copied.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:174]: Excellent! Now, who do I collect my $5 from?

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:236]: Not all, since "Access" is a header and not a link, but I will update "API Keys".

Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:236]: I didn't want to add a link since it will be different depending on your colo. I navigated to it from the user menu:

image

But is "User Menu" the right generic term for that?

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:391]: It might be good, but it looks like that would require extra configuration to format the data so it could be separated by type, and I'd rather not go that deep into documenting a third party plugin written for another third party tool.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:393]: That's from [this issue](https://github.com/gravitational/teleport-plugins/issues/747). I'm hoping it will be resolved soon, since it affects all guides using the generated configuration.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:165]: Resolved in a72c62d1ffe6995a82a7dda8a8ee8cfadd846751

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx]: Resolved in a72c62d1ffe6995a82a7dda8a8ee8cfadd846751

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:73]: Resolved in a72c62d1ffe6995a82a7dda8a8ee8cfadd846751

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:77]: Resolved in a72c62d1ffe6995a82a7dda8a8ee8cfadd846751

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:232]: Resolved in a72c62d1ffe6995a82a7dda8a8ee8cfadd846751

File: docs/pages/management/export-audit-events/elastic-stack.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/elastic-stack.mdx:125]: Resolved in fad1a0a6a29ee90e6baaff39b4dae4a6767c6ed9

File: docs/pages/management/export-audit-events/splunk.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/splunk.mdx:132]: Resolved in b097a42ae92d40695296cd3b47603563832ef0f5

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (ptgott) [reply docs/pages/management/export-audit-events/datadog.mdx:386]: As far as I know, Datadog should be automatically parsing the JSON documents the plugin is sending to it—is that not the case in this setup?

File: docs/config.json
Comment (zmb3) [reply docs/config.json:575]: I prefer export if you're happy with either.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:73]: Updated.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:386]: Nope, doesn't seem to be.

File: docs/pages/management/export-audit-events/elastic-stack.mdx
Comment (ptgott) [inline docs/pages/management/export-audit-events/elastic-stack.mdx:2]: Logstash is exporting audit events while Elasticsearch is ingesting/storing/indexing them. Is there another way to describe this?

File: docs/pages/management/export-audit-events/splunk.mdx
Comment (ptgott) [inline docs/pages/management/export-audit-events/splunk.mdx:2]: The Splunk forwarder is exporting, but Splunk itself is collecting, indexing, etc.

**Review by ptgott — COMMENTED** [review]
This review is more of a copy review, while last time I focused on testing.

File: docs/config.json
Comment (alexfornuto) [reply docs/config.json:575]: Export was chosen by @zmb3 in his edit.

File: docs/pages/includes/install-event-handler.mdx
Comment (alexfornuto) [reply docs/pages/includes/install-event-handler.mdx:9]:
```suggestion
We currently only build the Event Handler plugin for amd64 machines. For ARM
```

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:10]: In those setups we connect Event Handler to _some_ tool that's part of those ecosystems, as opposed to a third-party like Fluentd.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:54]: Oops, should have caught that myself after my first writing run-through.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:77]: Added a docker tab to the configure partial.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:197]: I'd like to push back on this a bit. I know what our style guide says, but putting myself in the position of a reader, I'd like an opportunity to review a piece of third party software instead of blindly installing in into my system.

Also consider the future where the instructions below might go stale, and until we know/resolve it the first party docs should (finger crossed) update with changes.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:77]: ....and removed it, because it seems that you can't run the configure command from outside the container.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:216]: The current working directory is where the certs are when you run the `configure` command. Creating a specific directory means additional host-side instructions that may or may not be relevant to the end user. Mounting each key individually adds a lot of complexity to the commands.

These are examples. at some point we have to trust the reader to know enough about their own environment and common tools like Docker to be able to adjust as needed for themselves.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:383]: Moving.

File: docs/pages/management/export-audit-events/elastic-stack.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/elastic-stack.mdx:2]: I will again refer to @zmb3 who suggested this term for all these docs.

File: docs/pages/management/export-audit-events/splunk.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/splunk.mdx:2]:
```suggestion
title: Export Teleport Audit Events to Splunk
```

File: docs/config.json
Comment (alexfornuto) [reply docs/config.json]: Done.

File: docs/config.json
Comment (ptgott) [reply docs/config.json:575]: Would "Export ... to Datadog" work for you?

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (ptgott) [inline docs/pages/management/export-audit-events/datadog.mdx:12]:
```suggestion
to establish a secure channel. In this setup, the Event Handler sends events to Fluentd, which forwards them to Datadog using an API key to authenticate.
```
How about something like this? I don't think we need to know how the Splunk/Elastic setups work in order to use this guide, and this describes the way Fluentd works here.

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (ptgott) [reply docs/pages/management/export-audit-events/datadog.mdx:77]: 👍 Would it make sense to change the "The instructions below assume a local Docker container for testing" note in the Prerequisites to, "The instructions below assume that you are running the Event Handler on your workstation" or something like that, so we're not tied to the Docker approach?

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (ptgott) [reply docs/pages/management/export-audit-events/datadog.mdx:197]: In that case, would it make sense to tell the reader to follow the installation instructions in the `fluent-plugin-datadog` README?

File: docs/pages/management/export-audit-events/elastic-stack.mdx
Comment (ptgott) [reply docs/pages/management/export-audit-events/elastic-stack.mdx:2]: Maybe "Export ... to the Elastic Stack"?

File: docs/config.json
Comment (alexfornuto) [reply docs/config.json:575]: I love it! The only way it could be better is if it was already in a suggestion, so I could have just clicked a button to make it happen.

```suggestion
              "title": "Export Audit Events to Datadog",
```

File: docs/pages/management/export-audit-events/datadog.mdx
Comment (alexfornuto) [reply docs/pages/management/export-audit-events/datadog.mdx:197]: Not for the other reader who _doesn't_ want to review their documentation, or already has and is following this guide to recreate on another environment, etc. This is a middle-ground where the user can _choose_ to follow the link but doesn't have to, and our docs contain a reasonable amount of content for a third party system.

File: docs/pages/management/export-audit-events/elastic-stack.mdx
Comment (alexfornuto) [inline docs/pages/management/export-audit-events/elastic-stack.mdx:2]:
```suggestion
title: "Export Teleport Audit Events with to Elastic Stack"
```

File: docs/config.json
Comment (alexfornuto) [inline docs/config.json:580]:
```suggestion
              "title": "Export Audit Events to the Elastic Stack",
```

**Review by ptgott — APPROVED** [review]

**Pull request merged**
